doc/devguide/extending/detect/app-layer-frames.rst

```diff
@@ -0,0 +1,173 @@
+*******************************
+Application Layer Frame Support
+*******************************
+
+.. contents:: Table of Contents
+
+Baseline
+========
+
+- Basic functioning of the Suricata engine, like application layer `transactions <https://github.com/OISF/suricata/blob/master/doc/devguide/extending/app-layer/transactions.rst>`_ and Suricata flows
+- `Suricata rules format <https://suricata.readthedocs.io/en/latest/rules/intro.html>`_
+- `Multi-pattern rule matching <https://suricata.readthedocs.io/en/latest/configuration/suricata-yaml.html?highlight=multi%20pattern%20matcher#detection-engine>`_ algorithm, used for rule filtering and to address engine performance
+
+General Concepts
+================
+
+.. note:: Stream Frames support is a work in progress. This means that names and some implementation details might still change, before we reach a more stable version of the code. The overall idea behind it is matured, though, so this documentation is still valid. Current progress: `Feat/appl records <https://github.com/OISF/suricata/pull/6713>`_
+
+Suricata's architecture and current state of parsers mean that the network traffic stream isn't readily available to the
+detection engine. Except for the application layer parsers, the Suricata engine itself only has access to higher level
+abstractions - *State* and *Transactions* (much heavier objects, that store *Frame* information in the form of *Request-Response* pairs).
+
+.. note:: For Suricata, *Frame* is a generic enough term that it can represent any unit of network data we are interested in, which could be comprised of one or several packages of other, lower level protocol(s).
+
```

Comment:
Not sure if `note` is good, or if should just keep as normal text. Also not sure if this definition is good enough.